lib/phoenix/endpoint.ex

```diff
@@ -411,6 +411,10 @@ defmodule Phoenix.Endpoint do
     quote do
       @behaviour Phoenix.Endpoint
 
+      def __start_sockets__ do
+        unquote(Keyword.get(opts, :start_sockets, []))
+      end
```

Comment:
I don't know yet if I really like having it explicit.

Reply:
Yeah, I'm with you. On the one hand it is really nice how this used to work. 

On the other hand I don't like the amount of stuff that's being configured on the endpoint socket macro. Because there's just a single entry point to "everything" that also means everything needs to somehow be configured right there.

Given routers and endpoints are really only quite loosely coupled there's afaik also no good way to retain the static knowledge of to be started socket modules from the router macro.